libnvme/src/logpage.rs

```diff
@@ -106,6 +100,13 @@ fn get_logpage_size(
                 },
             )?;
 
+            // Clean up the temporary req output so that it's not left with a
+            // dangling pointer after this function returns.
```

Comment:
I was going to say that if either of the calls in between `nvme_log_req_set_output` and this cleanup fail, we'll leave `req` with a dangling pointer. But I think that's the covered by the `// NB: if this call fails it's important that we drop req` comment in `Controller::get_logpage()`, right?

I wonder if it's worth it to enforce this statically. If we took an owned `NvmeLogReq` instead of `&mut`, _we_ would to drop the `req` if either of those calls fails. We'd have to return `Result<(NvmeLogReq, usize), Error>`, which is a little ugly, but maybe worth being able to drop the comment in `get_logpage()` that's important for safety but assumes implementation details of this function?

Reply:
Fixed in [f98d1f1](https://github.com/oxidecomputer/libnvme/pull/10/commits/f98d1f1bae1e0a4046f44c6ab88a390ae0cdd34c)